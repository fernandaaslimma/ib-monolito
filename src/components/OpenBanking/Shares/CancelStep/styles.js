import styled from "styled-components";
import { gray200, gray300 } from "../../../../styles/settings";
import { rem, remFontSize } from "../../../../styles/tools";

export const ContainerWrapper = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  max-width: 600px;
  margin: auto;
`;

export const CancelStepWrapper = styled.div`
  display: flex;
  flex-direction: column;

  justify-content: center;

  padding: ${rem(8)};

  max-width: ${rem(600)};
  margin: 0 auto;
`;

export const IconWrapper = styled.section`
  display: flex;
  justify-content: center;
  align-items: center;

  margin-top: ${rem(50)};
`;

export const CancelTitle = styled.span`
  font-family: Lato;

  font-style: normal;
  font-weight: bold;
  font-size: ${remFontSize(24)};
  line-height: 125%;
  /* identical to box height, or 30px */

  text-align: center;

  color: ${gray300};

  margin-top: ${rem(18)};
`;

export const CancelMessage = styled.span`
  font-family: Lato;

  font-style: normal;
  font-weight: normal;
  font-size: ${remFontSize(16)};
  line-height: 129%;

  text-align: center;

  color: ${gray200};

  margin: ${rem(24)} 0 ${rem(90)};
`;
