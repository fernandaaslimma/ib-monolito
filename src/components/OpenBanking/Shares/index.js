import Shares from "./Shares";
import { connect } from "redux-zero/react";
import { combineActions } from "redux-zero/utils";
import openBankingActions from "../../../actions/openBanking";

export default connect(
  ({
    transmittedCurrentShares,
    receivedCurrentShares,
    userInfo,
    error,
    consentCreated,
    shareResponseJsonPatch,
    institutions
  }) => ({
    transmittedCurrentShares,
    receivedCurrentShares,
    userInfo,
    error,
    consentCreated,
    shareResponseJsonPatch,
    institutions
  }),
  combineActions(openBankingActions)
)(Shares);
