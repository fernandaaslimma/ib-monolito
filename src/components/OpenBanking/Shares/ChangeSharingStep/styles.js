import styled, { css } from "styled-components";
import {
  gray200,
  gray300,
  white,
  neutral200
} from "../../../../styles/settings";
import { rem, remFontSize } from "../../../../styles/tools";

export const Description = styled.div`
  font-family: Lato;
  font-size: ${remFontSize(14)};
  font-weight: 400;
  font-style: normal;
  letter-spacing: ${rem(0.45)};
  line-height: ${rem(22, 4)};
  color: #587485;
`;

export const DescriptionEnd = styled(Description)`
  margin-top: ${rem(20)};
`;

export const ReceivingInstitutionInfo = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  margin-top: ${rem(16)};
`;

export const Wrapper = styled.div``;

export const Separator = styled.div`
  height: ${rem(1)};
  background-color: #d9e0e4;
`;

export const LinkDetails = styled.div`
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  justify-content: center;
  font-family: Lato;
  font-weight: 400;
  font-style: normal;
  font-size: ${remFontSize(14)};
  line-height: ${rem(16.8)};
  margin-top: ${rem(16)};
  margin-bottom: ${rem(6)};

  ${({ flexStart }) => {
    flexStart &&
      css`
        align-items: flex-start;
        margin-bottom: ${rem(16)};
      `;
  }}
`;

export const LinkCitizen = styled.span`
  text-decoration: underline;
  transition: 0.2s color;
  color: ${neutral200};
  cursor: pointer;
`;

export const SubTitle = styled.div`
  font-size: ${remFontSize(12)};
  margin: ${rem(16)} 0 ${rem(11.2)} ${rem(3)};
  font-weight: 500;
  font-style: normal;
  line-height: ${rem(14.4)};
`;

export const AccountBalance = styled.div`
  display: flex;
  background: #eef1f3;
  box-sizing: border-box;
  border-radius: ${rem(4)};
  align-items: center;
  align-content: center;
  border: ${rem(2)} #eef1f3 solid;
  cursor: pointer;
  margin-top: ${rem(16)}
    ${({ cursor }) =>
      cursor &&
      css`
        cursor: pointer;
      `}
    ${({ noSpacebetween }) =>
      !noSpacebetween &&
      css`
        background: none;
        justify-content: space-between;
      `};
`;

export const AccountText = styled.span`
  font-family: Lato;
  font-style: normal;
  font-weight: 400;

  font-size: ${remFontSize(15)};
  line-height: ${rem(20)};
  margin: ${rem(14)} 0 ${rem(14)} ${rem(15)};
  padding: 0;
  color: ${gray200};
`;

export const Value = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  margin: ${rem(12)} ${rem(10)} ${rem(11)} 0;
  padding: 0;
`;

export const SaveAccount = styled.div`
  display: flex;
  align-items: center;
  border-radius: ${rem(6)};
  margin: 0 ${rem(16)} ${rem(16)} ${rem(16)};
  padding: ${rem(15)};

  ${({ jContent }) =>
    jContent &&
    css`
      justify-content: ${jContent};
    `}

  ${({ checked }) =>
    checked &&
    css`
      background: #eef1f3;
    `}
`;

export const SaveAccountInfo = styled.div`
  display: flex;
  align-items: center;
  margin-left: ${rem(16)};
`;

export const AllText = styled.span`
  font-family: Lato;
  font-size: ${remFontSize(12.64)};
  font-weight: 500;
  font-style: normal;
  letter-spacing: ${rem(0.2)};
  margin-right: ${rem(16)};
`;

export const WrapperInstitution = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
`;

export const WapperImg = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  height: ${rem(36)};
  width: ${rem(36)};
  border-radius: ${rem(6)};
  border: 1px solid #d9e0e4;
`;

export const ReceivingInstitutionName = styled.span`
  font-family: Lato;
  font-style: normal;
  font-weight: bold;
  font-size: ${remFontSize(14)};
  line-height: 125%;
  letter-spacing: 0.452308px;
  color: ${gray300};
  margin-left: ${rem(16)};
`;

export const DescriptionNames = styled.div`
  font-family: Lato;
  font-size: ${remFontSize(14)};
  font-weight: 400;
  font-style: normal;
  letter-spacing: ${rem(0.45)};
  line-height: ${rem(22, 4)};
  color: #587485;
  margin-bottom: ${rem(40)};
`;

export const WrapperSearch = styled.div`
  margin: ${rem(16)} ${rem(16)} ${rem(24)} ${rem(16)};
`;

export const CardAccounts = styled.div`
  font-family: Lato;
  font-style: normal;
  padding: ${rem(16)};
  margin-bottom: ${rem(16)};
  background: ${white};
  border: 1px solid #eef1f3;
  box-shadow: 0px 1px 2px rgba(0, 0, 0, 0.08);
  border-radius: 6px;

  ${({ padding }) =>
    padding &&
    css`
      padding: ${rem(padding)} ${rem(16)};
    }
  `}
`;

export const StickyWrapper = styled.div`
  overflow: auto;
  position: sticky;
  bottom: 0;
  background: ${white};
  border-top: solid ${rem(1)} #d3dde4;
`;

export const ContainerWrapper = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  max-width: 600px;
  margin: auto;
`;

export const BtnWrapper = styled.div`
  flex-grow: 1;
  position: relative;
  flex-basis: 0;
  display: flex;

  > div {
    flex-grow: 1;
    flex-basis: 0;
  }
`;

export const Title = styled.div`
  font-family: Lato;
  font-size: ${remFontSize(12)};
  font-weight: 400;
  font-style: normal;
  letter-spacing: ${rem(0.45)};
  line-height: ${rem(15)};
  color: #587485;
  margin-top: ${rem(16)};
`;

export const Text = styled.span`
  font-family: Lato;
  font-size: ${remFontSize(16)};
  font-weight: 600;
  font-style: normal;
  color: #2d4758;
  margin-right: ${rem(16)};
`;

export const SaveAccountDeadLine = styled.div`
  display: flex;
  align-items: center;
  border-radius: ${rem(6)};
  margin: 0 ${rem(16)} ${rem(16)} ${rem(16)};
  padding: ${rem(15)};

  ${({ jContent }) =>
    jContent &&
    css`
      justify-content: ${jContent};
    `}

  ${({ checked }) =>
    checked &&
    css`
      background: #eef1f3;
    `}
`;

export const SaveAccountInfoDeadLine = styled.div`
  display: flex;
  align-items: center;
  margin-left: ${rem(16)};
`;
