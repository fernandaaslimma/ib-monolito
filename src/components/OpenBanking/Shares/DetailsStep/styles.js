import styled, { css } from "styled-components";
import { gray300, white, gray200, grey40 } from "../../../../styles/settings";
import { rem, remFontSize } from "../../../../styles/tools";

export const SubTitle = styled.h2`
  font-size: ${remFontSize(16)};
  margin: 0 0 ${rem(15)} 0;
  color: ${gray300};
  font-weight: 400;
  line-height: ${rem(24)};
  letter-spacing: 0.452308px;
`;

export const BtnWrapper = styled.div`
  flex-grow: 1;
  position: relative;
  flex-basis: 0;
  display: flex;

  > div {
    flex-grow: 1;
    flex-basis: 0;
  }
`;

export const StickyWrapper = styled.div`
  overflow: auto;
  position: sticky;
  bottom: 0;
  background: ${white};
  border-top: solid ${rem(1)} #d3dde4;
`;

export const InstitutionInfo = styled.section`
  display: flex;
  align-items: center;
  margin: ${rem(14)} 0 0 0;
`;

export const InstitutionName = styled.span`
  font-family: Lato;
  font-style: normal;
  font-weight: bold;
  font-size: ${remFontSize(14)};
  line-height: 125%;
  /* or 17px */

  letter-spacing: 0.452308px;
  color: ${gray300};
`;
export const ShareDetails = styled.div`
  display: flex;
  flex-direction: column;
  align-content: flex-end;
  margin-left: ${rem(16)};
`;
export const SharedDate = styled.span`
  font-family: Lato;
  font-style: normal;
  font-weight: 600;
  font-size: ${remFontSize(10)};
  line-height: 125%;
  /* identical to box height, or 12px */

  letter-spacing: 0.45px;

  color: ${gray200};

  margin-top: ${rem(8)};
`;

export const WrapperTag = styled.div`
  display: flex;
  justify-content: flex-end;
  align-items: center;
`;

export const EndSharringSheetWrapper = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  height: ${rem(500)};

  @media (max-width: ${rem(980)}) {
    height: calc(100vh - 3.25rem);
  }
`;

export const InfoWrapper = styled.div`
  display: flex;
  flex-direction: column;
  padding: ${rem(48)} ${rem(16)} ${rem(24)} ${rem(16)};

  align-items: center;
`;

export const MainMessage = styled.div`
  font-family: Lato;
  font-style: normal;
  font-weight: bold;
  font-size: ${remFontSize(20)};
  line-height: ${rem(30)};
  text-align: center;
  letter-spacing: 0.416667px;
  color: #2d4758;
  margin: ${rem(32)} 0 ${rem(40)} 0;
`;

export const ButtonWrapper = styled.div`
  margin-top: auto;
  overflow: auto;
  bottom: 0;
  padding: 0 ${rem(6)} ${rem(24)};
  display: grid;
  grid-template-columns: auto auto;

  @media (max-width: ${rem(980)}) {
    position: sticky;
    grid-template-columns: none;
  }
`;

export const ButtonWrapperClose = styled.div`
  padding: 0 ${rem(6)} ${rem(24)};
  display: grid;
  grid-template-columns: auto;
`;

export const Message = styled.div`
  font-family: Lato;
  font-style: normal;
  font-weight: normal;
  font-size: ${remFontSize(16)};
  line-height: ${rem(24)};
  display: inline;
  text-align: center;
  letter-spacing: 0.426667px;
  color: ${grey40};
`;

export const MessageBold = styled(Message)`
  font-family: Lato;
  font-weight: 500;
  font-style: normal;
  letter-spacing: ${rem(0.45)};
  line-height: ${rem(15)};
  font-weight: 700;

  ${({ color }) =>
    color &&
    css`
      color: ${color};
    `}
`;

export const ChangeSharringSheetWrapper = styled.p`
  font-family: Lato;
  font-style: normal;
  font-weight: normal;
  padding: ${rem(24)} ${rem(16)};
  font-size: ${remFontSize(14)};
  line-height: 150%;
  color: ${gray200};
`;

export const Bold = styled.span`
  font-weight: bold;
`;

export const ChangeSharringButtonWrapper = styled.div`
  padding: 0 ${rem(6)} ${rem(24)};
  display: grid;
  grid-template-columns: auto auto;
`;
