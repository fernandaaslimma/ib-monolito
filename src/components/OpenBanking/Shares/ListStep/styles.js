import styled from "styled-components";
import { gray200, gray300, white } from "../../../../styles/settings";
import { rem, remFontSize } from "../../../../styles/tools";

export const Wrapper = styled.div`
  padding: ${rem(10)} 0 ${rem(32)} 0;
  max-width: ${rem(600)};
  margin: 0 auto;
  cursor: pointer;
  @media (max-width: ${rem(600)}) {
    padding: ${rem(10)} ${rem(16)} ${rem(32)} ${rem(16)};
  }
`;

export const UpperWrapper = styled.div`
  max-width: ${rem(600)};
  margin: 0 auto;
`;

export const Card = styled.section`
  background: #ffffff;
  display: flex;
  margin-top: ${rem(16)};
  justify-content: space-between;
  padding: ${rem(16)} 0;

  border: 1px solid #eef1f3;
  box-sizing: border-box;
  box-shadow: 0px 1px 2px rgba(0, 0, 0, 0.16);
  border-radius: 6px;
  min-width: 263px;

  :hover {
    cursor: pointer;
  }
`;

export const InstitutionInfo = styled.section`
  display: flex;
  align-items: center;
  margin-left: ${rem(16)};
`;

export const InstitutionName = styled.span`
  font-family: Lato;
  font-style: normal;
  font-weight: bold;
  font-size: 14px;
  line-height: 125%;
  /* or 17px */

  letter-spacing: 0.452308px;

  color: #2d4758;

  margin-left: ${rem(16)};
`;
export const ShareDetails = styled.div`
  display: flex;
  flex-direction: column;
  align-items: flex-end;
  margin-right: ${rem(12)};
`;
export const SharedDate = styled.span`
  white-space: nowrap;
  font-family: Lato;
  font-style: normal;
  font-weight: 600;
  font-size: 10px;
  line-height: 125%;
  /* identical to box height, or 12px */
  letter-spacing: 0.45px;
  color: #587485;
  margin-top: ${rem(19)};
`;

export const InfoCard = styled.div`
  padding: ${rem(16)} ${rem(15)} ${rem(24)} ${rem(15)};
  background: ${white};
  overflow: auto;
  font-family: Lato;
  font-size: ${remFontSize(14)};
  color: ${gray200};
  line-height: ${remFontSize(22)};
`;

export const Title = styled.h2`
  font-size: ${remFontSize(18)};
  color: ${gray300};
  font-weight: 700;
  line-height: 125%;
  /* identical to box height, or 20px */

  letter-spacing: 0.452308px;
`;

export const AccessPortal = styled.span`
  cursor: pointer;
  margin-bottom: ${rem(40)};
  display: flex;
  justify-content: center;
  font-family: Lato;
  font-style: normal;
  font-weight: bold;
  font-size: 14px;
  line-height: 17px;
  /* identical to box height */

  text-align: center;
  letter-spacing: 0.2px;

  color: #3976cf;
`;

export const WapperBankImg = styled.img`
  object-fit: cover;
  height: ${rem(36)};
  width: ${rem(36)};
  border-radius: ${rem(6)};
`;
