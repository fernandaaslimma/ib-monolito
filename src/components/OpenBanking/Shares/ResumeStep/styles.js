import styled, { css } from "styled-components";
import { white, neutral200, gray200 } from "../../../../styles/settings";
import { rem, remFontSize } from "../../../../styles/tools";

export const BtnWrapper = styled.div`
  flex-grow: 1;
  position: relative;
  flex-basis: 0;
  display: flex;

  > div {
    flex-grow: 1;
    flex-basis: 0;
  }
`;

export const StickyWrapper = styled.div`
  overflow: auto;
  position: sticky;
  bottom: 0;
  background: ${white};
  border-top: solid ${rem(1)} #d3dde4;
`;

export const ContentWrapper = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  padding: 0 ${rem(16)};
`;

export const ContentTitle = styled.span`
  font-family: Lato;
  font-weight: 400;
  font-style: normal;
  font-size: ${remFontSize(14)};
  line-height: ${rem(22.4)};
  letter-spacing: ${rem(0.45)};
`;
export const Content = styled.div`
  display: flex;
  flex-direction: column;
  margin: ${rem(35)} 0 ${rem(40)} 0;
`;

export const Data = styled.span`
  font-family: Lato;
  font-weight: 700;
  font-style: normal;
  font-size: ${rem(14)};
  line-height: ${rem(16.8)};
  margin-top: ${rem(12)};
`;

export const LinkDetails = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-family: Lato;
  font-weight: 400;
  font-style: normal;
  font-size: ${rem(14)};
  line-height: ${rem(16.8)};
`;

export const LinkCitizen = styled.span`
  cursor: pointer;
  transition: 0.2s color;
  color: ${neutral200};
`;

export const AccountBalance = styled.div`
  display: flex;
  justify-content: space-between;
  box-sizing: border-box;
  border-radius: ${rem(4)};
  align-items: center;
  align-content: center;
  border: ${rem(2)} #eef1f3 solid;

  ${({ cursor }) =>
    cursor &&
    css`
      cursor: pointer;
    `}
`;

export const Value = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  margin: ${rem(12)} ${rem(10)} ${rem(11)} 0;
  padding: 0;
`;

export const AccountText = styled.span`
  font-family: Lato;
  font-style: normal;
  font-weight: 400;

  font-size: ${remFontSize(15)};
  line-height: ${rem(20)};
  margin: ${rem(14)} 0 ${rem(14)} ${rem(15)};
  padding: 0;
  color: ${gray200};
`;

export const ResourceInfo = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  font-family: Lato;
  font-size: ${remFontSize(14)};
  font-weight: 500;
  font-style: normal;
  line-height: ${remFontSize(17, 5)};
  margin-right: ${rem(16)};
  align-items: center;
  color: #587485;
  margin: ${rem(16)} ${rem(16)} ${rem(20)} ${rem(16)};
`;

export const SaveAccount = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  border-radius: ${rem(6)};
  margin: 0 ${rem(16)} ${rem(12)} ${rem(16)};
  padding: ${rem(15)};

  ${({ jContent }) =>
    jContent &&
    css`
      justify-content: ${jContent};
    `}

  ${({ checked }) =>
    checked &&
    css`
      background: #eef1f3;
    `}
`;
export const SaveAccountInfo = styled.div`
  display: flex;
  align-items: center;
`;

export const Text = styled.span`
  font-family: Lato;
  font-size: ${remFontSize(16)};
  font-weight: 600;
  font-style: normal;
  color: #2d4758;
  margin-right: ${rem(16)};
`;

export const RadioButtonsWrapper = styled.div`
  display: flex;
  padding: ${rem(12)};

  @media (min-width: ${rem(348)}) {
    label + label {
      margin-left: ${rem(12)};
      padding-bottom: ${rem(32)};
    }
  }

  @media (max-width: ${rem(348)}) {
    flex-direction: column;
    label {
      margin-bottom: ${rem(23)};
    }
  }
`;

export const ResourceWrapper = styled.div`
  margin: 0 0 ${rem(8)} 0;
`;
