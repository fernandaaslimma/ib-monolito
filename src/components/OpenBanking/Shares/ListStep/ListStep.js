import React, { useContext } from "react";
import { Icon } from "react-bocombbm-components";
import { Image } from "../../../common/Image/Image";
import DefaultContent from "../../../common/DefaultContent";
import Tabs from "../../../common/Tabs/Tabs";
import Tag from "../../../common/Tag";
import { scrollToTop } from "../../../../utils/dom";
import DefaultShimmerLoading from "../../../common/DefaultShimmerLoading";
import { rem } from "../../../../styles/tools";
import { black30 } from "../../../../styles/settings";
import { translate } from "../../../../utils/i18n";
import { formatDate } from "../../../../utils/openBankingUtils";
import { OpenBankingSharesContext } from "../Shares";
import BankPng from "../../../common/Icon/default_bank100.png";
import { checkIfHasAccess } from "../../../common/CanAccess/CanAccess";
import { APPROVE_CONSENTS } from "../../../../utils/constants";

import {
  Wrapper,
  Card,
  InstitutionName,
  ShareDetails,
  SharedDate,
  InstitutionInfo,
  Title,
  InfoCard,
  UpperWrapper,
  AccessPortal
} from "./styles";

function ListStep({ stepForward }) {
  const {
    state: { selectedTab, loadingCurrentShares },
    props: {
      transmittedCurrentShares,
      receivedCurrentShares,
      institutions,
      userInfo
    },
    tagConf,
    setSelectedTab,
    setSelectedBank
  } = useContext(OpenBankingSharesContext);


  React.useEffect(() => {
    scrollToTop();
  }, []);

  const list = () => {
    const relativeShares =
      selectedTab === 0 ? receivedCurrentShares : transmittedCurrentShares;

    // const sharesInProgress = relativeShares.filter(
    //   share => share.status === "ACTIVE" || share.status === "PENDING"
    // );

    relativeShares.sort(function (a, b) {
      if (new Date(a.lastStatusUpdate) > new Date(b.lastStatusUpdate)) {
        return -1;
      } else {
        return true;
      }
    });

    const sharesInProgressWithLogo =
      selectedTab === 0
        ? relativeShares
        : relativeShares.map(share => {
          const institution = institutions.find(
            institution => institution.OrganisationId === share.organisationId
          );

          return {
            ...share,
            authorisationServer: {
              customerFriendlyLogoUri: institution?.CustomerFriendlyLogoUri || BankPng
            }
          };
        });

    return sharesInProgressWithLogo;
  };


  const PersonImage = ({ profileImageUrl, ...rest }) => (
    <Image src={profileImageUrl} srcOnError={BankPng} {...rest} />
  );

  const renderInstitutions = () => {
    return loadingCurrentShares ? (
      <DefaultShimmerLoading repeat={3} innerRepeat={2} />
    ) : list().length > 0 ? (
      <Wrapper>
        {list().map((item, index) => {
          return (
            <Card
              data-test="card"
              key={index}
              onClick={() => setSelectedBank(item, stepForward)}
            >
              <InstitutionInfo>
                {item && (
                  <PersonImage
                    profileImageUrl={
                      item.authorisationServer.customerFriendlyLogoUri
                    }
                  />
                )}
                <InstitutionName>
                  {item.organisationName ||
                    item.authorisationServer.customerFriendlyName}
                </InstitutionName>
              </InstitutionInfo>
              <ShareDetails>
                {item.status && (
                  <Tag
                    dataTest={`status_${index}`}
                    title={tagConf[item.status.toLowerCase()][2]}
                    color={tagConf[item.status.toLowerCase()][0]}
                    titleColor={tagConf[item.status.toLowerCase()][1]}
                  />
                )}
                <SharedDate>{formatDate(item.lastStatusUpdate)}</SharedDate>
              </ShareDetails>
            </Card>
          );
        })}
      </Wrapper>
    ) : (
      <Wrapper>
        <DefaultContent
          data-test="emptyStatements"
          Icon={() => (
            <Icon
              type={"NoTransactions"}
              color={black30}
              height={rem(66)}
              width={rem(66)}
            />
          )}
          primaryText={translate("OPEN_BANKING_LIST_NO_SHARINGS_MSG_1")}
          secondaryTexts={[translate("OPEN_BANKING_LIST_NO_SHARINGS_MSG_2")]}
        />
      </Wrapper>
    );
  };

  const openCitizenPortal = () => {
    window.open("https://openbankingbrasil.org.br/", "_blank");
  };

  const hasAproveConsents = checkIfHasAccess(userInfo, [APPROVE_CONSENTS]);

  return (
    <UpperWrapper>
      <InfoCard>
        <Title>{translate("OPEN_BANKING_MY_SHARES")}</Title>
      </InfoCard>

      <Tabs selectedTab={selectedTab} setSelectedTab={setSelectedTab}>
        <section title={translate("OPEN_BANKING_RECEIVED")}>
          {renderInstitutions()}
        </section>
        <section
          hide={!hasAproveConsents}
          title={translate("OPEN_BANKING_TRANSMITTED")}
        >
          {renderInstitutions()}
        </section>
      </Tabs>

      <AccessPortal onClick={() => openCitizenPortal()}>
        {translate("OPEN_BANKING_ACCESS_CITIZEN_PORTAL")}
      </AccessPortal>
    </UpperWrapper>
  );
}

export default ListStep;
