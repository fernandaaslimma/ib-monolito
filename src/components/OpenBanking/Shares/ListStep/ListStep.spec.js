import React from "react";
import { shallow } from "enzyme";
import ListStep from "./ListStep";
import { Card, AccessPortal } from "./styles";
import { OpenBankingSharesContext } from "../Shares";

delete window.open;

window.open = jest.fn();

let context = {
  state: { selectedTab: 1, loadingCurrentShares: false },
  props: {
    institutions: [
      {
        CustomerFriendlyLogoUri:
          "https://upload.wikimedia.org/wikipedia/commons/b/b9/Slack_Technologies_Logo.svg",
        OrganisationId: "mock id"
      }
    ],
    transmittedCurrentShares: [
      {
        organisationId: "mock id",
        authorisationServer: null,
        lastStatusUpdate: "2021-11-03T19:16:01.565Z",
        organisationName: "BCO BOCOM BBM S.A.",
        status: "ACTIVE"
      }
    ],
    receivedCurrentShares: [
      {
        organisationId: "mock id",
        authorisationServer: {
          customerFriendlyLogoUri:
            "https://upload.wikimedia.org/wikipedia/commons/b/b9/Slack_Technologies_Logo.svg"
        },
        customerFriendlyName: "BOCOM BBM",
        lastStatusUpdate: "2021-11-03T19:16:01.565Z",
        organisationName: "BCO BOCOM BBM S.A.",
        status: "PENDING"
      }
    ]
  },
  tagConf: {
    inactive: ["#D9E0E4", "#244859", "Desativado"],
    expired: ["#D9E0E4", "#244859", "Expirado"],
    pending: ["#FFD46A", "#80521B", "Pendente"],
    active: ["#CCE9E1", "#004933", "Ativo"]
  },
  formatDate: jest.fn(),
  setSelectedTab: jest.fn(),
  setSelectedBank: jest.fn(),
  capitalized: jest.fn()
};

const props = {
  stepForward: jest.fn()
};

describe("ListStep", () => {
  let useEffect;
  const mockUseEffect = () => {
    useEffect.mockImplementation(f => f());
  };
  beforeEach(() => {
    useEffect = jest.spyOn(React, "useEffect");
    mockUseEffect();
  });

  it("should match snapshot", () => {
    expect(
      shallow(
        <OpenBankingSharesContext.Provider value={context}>
          <ListStep {...props} />
        </OpenBankingSharesContext.Provider>
      )
    ).toMatchSnapshot();
  });

  it.skip("should call next step", () => {
    const component = shallow(<ListStep {...props} />);
    component
      .find(Card)
      .at(0)
      .simulate("click");
    expect(context.setSelectedBank).toHaveBeenCalled();
  });

  it.skip("should call citizen portal", () => {
    const component = shallow(<ListStep {...props} />);
    component.find(AccessPortal).simulate("click");
    expect(window.open).toHaveBeenCalled();
  });

  it("should match snapshot without list", () => {
    context.props.transmittedCurrentShares = [];
    expect(
      shallow(
        <OpenBankingSharesContext.Provider value={context}>
          <ListStep {...props} />
        </OpenBankingSharesContext.Provider>
      )
    ).toMatchSnapshot();
  });

  it("should match snapshot with received shares", () => {
    expect(
      shallow(
        <OpenBankingSharesContext.Provider value={context}>
          <ListStep {...props} />
        </OpenBankingSharesContext.Provider>
      )
    ).toMatchSnapshot();
  });

  it("should match snapshot with received shares", () => {
    context.state.loadingCurrentShares = true;
    expect(
      shallow(
        <OpenBankingSharesContext.Provider value={context}>
          <ListStep {...props} />
        </OpenBankingSharesContext.Provider>
      )
    ).toMatchSnapshot();
  });
});
