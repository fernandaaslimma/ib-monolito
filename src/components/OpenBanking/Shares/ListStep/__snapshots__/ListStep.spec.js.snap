// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListStep should match snapshot 1`] = `
<ListStep
  stepForward={[MockFunction]}
/>
`;

exports[`ListStep should match snapshot with received shares 1`] = `
<ListStep
  stepForward={[MockFunction]}
/>
`;

exports[`ListStep should match snapshot with received shares 2`] = `
<ListStep
  stepForward={[MockFunction]}
/>
`;

exports[`ListStep should match snapshot without list 1`] = `
<ListStep
  stepForward={[MockFunction]}
/>
`;